lib/rubocop/cop/lint/format_parameter_mismatch.rb

```diff
@@ -0,0 +1,78 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Lint
+      # This lint sees if there is a mismatch between the number of
+      # expected parameters for format and what is actually passed.
+      class FormatParameterMismatch < Cop
+
+        VALID_FIELDS_REGEX = /%[bBdiouxXeEfgGacps]/
+
+        def on_send(node)
+          add_offense(node, :selector) if offending_node?(node)
+        end
+
+        private
+
+        def offending_node?(node)
+          (sprintf?(node) || format?(node)) && missing_parameters?(node)
+        end
+
+        def format_method?(name, node)
+          receiver, method_name, *args = *node
+
+          # commands have no explicit receiver
+          unless !receiver && method_name == name
+            false
+          else
+            args.size > 1
+          end
+        end
+
+        def expected_fields(args)
+          args
+            .first
+            .loc
+            .expression
+            .source
+            .scan(VALID_FIELDS_REGEX)
+        end
+
+        def missing_parameters?(node)
+          _receiver, _method_name, *args = *node
+
+          number_of_args_for_format = (args.size - 1)
+          number_of_expected_fields = expected_fields(args).size
+
+          number_of_expected_fields != number_of_args_for_format
+        end
+
+        def format?(node)
+          format_method?(:format, node)
+        end
+
+        def sprintf?(node)
+          format_method?(:sprintf, node)
+        end
+
+        def percent?(node)
```

Comment:
I'd call this "format_specifier" or something like this. Probably there's an ever more formal term.